controllers/space/space_controller_test.go

```diff
@@ -1794,6 +1794,219 @@ func TestRetargetSpace(t *testing.T) {
+// and all the consequences and expectations of this link between two different spaces.
+func TestSubSpace(t *testing.T) {
+
+	// given
+	logf.SetLogger(zap.New(zap.UseDevMode(true)))
+	s := scheme.Scheme
+	err := apis.AddToScheme(s)
+	require.NoError(t, err)
+	basicTier := tiertest.BasicTier(t, tiertest.CurrentBasicTemplates)
+
+	// test SpaceBindings of a parentSpace are created/updated/deleted ,
+	// and how this should affect the subSpace and its NSTemplateSet
+	t.Run("SpaceBindings inheritance ", func(t *testing.T) {
+
+		t.Run("create parentSpace with admin and viewer roles, and expect subSpace will have same usernames and roles", func(t *testing.T) {
+			// given a Space and its NSTemplateSet resource...
+			parentSpace := spacetest.NewSpace("parentSpace",
+				spacetest.WithTierName(basicTier.Name),
+				spacetest.WithSpecTargetCluster("member-1"),
+				spacetest.WithStatusTargetCluster("member-1"), // already provisioned on a target cluster
+				spacetest.WithFinalizer())
+			parentNStmplSet := nstemplatetsettest.NewNSTemplateSet(parentSpace.Name,
+				nstemplatetsettest.WithReferencesFor(basicTier,
+					// include pre-existing users with role...
+					nstemplatetsettest.WithSpaceRole("admin", "parentAdmin"),
+					nstemplatetsettest.WithSpaceRole("viewer", "parentViewer"),
+				),
+				nstemplatetsettest.WithReadyCondition(),
+			)
+			// ...and their corresponding space bindings
+			sb1 := spacebindingtest.NewSpaceBinding("parentSpaceAdmin", parentSpace.Name, "admin", "signupAdmin")
+			sb2 := spacebindingtest.NewSpaceBinding("parentSpaceViewer", parentSpace.Name, "viewer", "signupViewer")
+
+			// ...now create a subSpace
+			subSpace := spacetest.NewSpace("subSpace",
+				spacetest.WithTierName(basicTier.Name),
+				spacetest.WithSpecTargetCluster("member-1"),     // already provisioned on a target cluster
+				spacetest.WithStatusTargetCluster("member-1"),   // already provisioned on a target cluster
+				spacetest.WithSpecParentSpace(parentSpace.Name), // set the parentSpace spec
+				spacetest.WithFinalizer())
+			subNStmplSet := nstemplatetsettest.NewNSTemplateSet(subSpace.Name,
+				nstemplatetsettest.WithReadyCondition(), // create empty NSTemplate set for subSpace
+			)
+
+			hostClient := test.NewFakeClient(t, parentSpace, sb1, sb2, basicTier, subSpace)
+			member1Client := test.NewFakeClient(t, parentNStmplSet, subNStmplSet)
+			member1 := NewMemberClusterWithClient(member1Client, "member-1", corev1.ConditionTrue)
+
+			ctrl := newReconciler(hostClient, member1)
+			InitializeCounters(t,
+				NewToolchainStatus(
+					WithMetric(toolchainv1alpha1.UserSignupsPerActivationAndDomainMetricKey, toolchainv1alpha1.Metric{
+						"1,internal": 1,
+					}),
+					WithMetric(toolchainv1alpha1.MasterUserRecordsPerDomainMetricKey, toolchainv1alpha1.Metric{
+						string(metrics.Internal): 1,
+					}),
+					WithMember("member-1", WithSpaceCount(2)),
+				))
+
```

Comment:
probably the same thing also for the counters. Feel free to keep it there if you think that it's important - just add a comment with a note to it

Reply:
Yeah, the counters were unchanged so there was not big value in there. It just makes the test less readable.
Removed in : [6b9e585](https://github.com/codeready-toolchain/host-operator/pull/733/commits/6b9e5853271c52ff603d67960dbfd63ba4f10904)